staging/src/k8s.io/kubectl/pkg/cmd/debug/debug.go

```diff
@@ -325,65 +333,67 @@ func (o *DebugOptions) Run(f cmdutil.Factory, cmd *cobra.Command) error {
 	}
 	o.podClient = clientset.CoreV1()
 
-	r := f.NewBuilder().
+	builder := f.NewBuilder().
 		WithScheme(scheme.Scheme, scheme.Scheme.PrioritizedVersionsAllGroups()...).
 		NamespaceParam(o.Namespace).DefaultNamespace().ResourceNames("pods", o.TargetNames...).
-		Do()
-	if err := r.Err(); err != nil {
-		return err
+		SingleResourceType().
+		Flatten()
```

Comment:
What's the reasoning behind SingleResourceType and Flatten?

Reply:
Just to guarantee that we are debugging only one resource at a time as we're discussing in here https://github.com/kubernetes/kubernetes/pull/111453#discussion_r931103797